src/Symfony/Component/DependencyInjection/Loader/PhpFileLoader.php

```diff
@@ -145,16 +145,8 @@ class_exists(ContainerConfigurator::class);
 
         $callback(...$arguments);
 
-        $this->loadFromExtensions($configBuilders);
-    }
-
-    /**
-     * @param iterable<ConfigBuilderInterface> $configBuilders
-     */
-    private function loadFromExtensions(iterable $configBuilders): void
-    {
         foreach ($configBuilders as $configBuilder) {
-            $this->loadExtensionConfig($configBuilder->getExtensionAlias(), $configBuilder->toArray());
+            $containerConfigurator->extension($configBuilder->getExtensionAlias(), $configBuilder->toArray(), $this->prepend);
```

Comment:
I still need to do some config prepending tests, but I feel that we've missed something here as the previous `$this->loadExtensionConfig()` is currently doing more things when prepend argument is `true`, e.g. the nested importing might be broken...

Reply:
I saw you made the `ContainerConfigurator` able to prepend in #52636 so I assumed it was okay but it is indeed missing this part: https://github.com/symfony/symfony/blob/7c956d84af558313657e25b39bf098281206cd5d/src/Symfony/Component/DependencyInjection/Loader/FileLoader.php#L238-L245

Don’t know if it is an issue at all though :sweat:

Reply:
Yes, it'll be an issue when want to prepend nested configs using imports

Reply:
If you already have the tests I think you’re more reading than I am; go ahead :grin: 

Would `ContainerConfiguration::extension` serve a purpose then?

Reply:
> Would ContainerConfiguration::extension serve a purpose then?

It should be avoided internally where recursive importing is involved, as now importing from `prependExtension()` is supported (which requires an specific holding strategy to keep the importing order). However, you can still use it on your end for trivial array config importing (either in the `prependExtension()` or `loadExtension()` methods).

Reply:
Okaaay thanks; glad I asked your review 😄